frontend/src/context/SubscriptionProvider.jsx

```diff
@@ -0,0 +1,181 @@
+      return {
+        ...currentState,
+        totalTransactions: action.totalTransactions,
+      };
+    case "accounts":
+      return {
+        ...currentState,
+        totalAccounts: action.totalAccounts,
+      };
+    case "dayTransactions":
+      return {
+        ...currentState,
+        lastDayTxCount: action.lastDayTxCount,
+      };
+    default:
+      return stateInit;
+  }
+};
+
+const SubContext = createContext({
+  finalTimestamp: 0,
+  dashState: stateInit,
+  nodeInfo: nodeInit,
+});
+
+export default (props) => {
+  const [dashState, dispatchState] = useReducer(stateReducer, stateInit);
+  const [nodeInfo, dispatchNode] = useReducer(nodeReducer, nodeInit);
+  const [finalTimestamp, dispatchFinalTimestamp] = useState(0);
+
+  // fetch total amount of blocks, txs and accounts and lastBlockHeight and txs for 24hr
+  const fetchNewStats = function (stats) {
+    // subsceiption data part
+    let states = stats[0].dataStats;
+    let [
+      lastBlockHeight,
+      totalAccounts,
+      totalBlocks,
+      totalTransactions,
+      lastDayTxCount,
+    ] = [
+      states.lastBlockHeight,
+      states.totalAccounts,
+      states.totalBlocks,
+      states.totalTransactions,
+      states.lastDayTxCount,
+    ];
+
+    // dispatch direct data part
+    if (dashState.lastBlockHeight !== lastBlockHeight) {
+      dispatchState({ type: "lastBlockHeight", lastBlockHeight });
+    }
+    if (dashState.totalAccounts !== totalAccounts) {
+      dispatchState({ type: "accounts", totalAccounts });
+    }
+    if (dashState.totalBlocks !== totalBlocks) {
+      dispatchState({ type: "blocks", totalBlocks });
+    }
+    if (dashState.totalTransactions !== totalTransactions) {
+      dispatchState({ type: "transactions", totalTransactions });
+    }
+    if (dashState.lastDayTxCount !== lastDayTxCount) {
+      dispatchState({ type: "dayTransactions", lastDayTxCount });
+    }
+  };
+
+  const fetchFinalTimestamp = (timestamp) => {
+    let final = timestamp[0];
+    if (finalTimestamp !== final) {
+      dispatchFinalTimestamp(final);
+    }
+  };
+
+  const fetchNodeInfo = (nodes) => {
+    let { onlineNodes, validators, proposals } = nodes[0];
+
+    let validatorAmount = validators.length;
+    let proposalAmount = proposals.length;
+    let onlineNodeAmount = onlineNodes.length;
+
+    if (nodeInfo.validatorAmount !== validatorAmount) {
+      dispatchNode({ type: "validators", validatorAmount, validators });
+    }
+    if (nodeInfo.onlineNodeAmount !== onlineNodeAmount) {
+      dispatchNode({ type: "onlines", onlineNodeAmount, onlineNodes });
+    }
+    if (nodeInfo.proposalAmount !== proposalAmount) {
+      dispatchNode({ type: "proposals", proposalAmount, proposals });
+    }
+  };
+
+  const Subscription = useCallback(() => {
+    new ExplorerApi().subscribe("data-stats", fetchNewStats);
+    new ExplorerApi().subscribe("final-timestamp", fetchFinalTimestamp);
+    new ExplorerApi().subscribe("nodes", fetchNodeInfo);
```

Comment:
Why did you choose to merge all of these into a single context? This way a user opening ANY page of Explorer will get flooded with irrelevant data (e.g. opening a Block details page one will still get online nodes and data-stats)

I suggest we split the context into smaller chunks. Currently, we have a monolith and I hardly see we can move forward with it adding new features.

Reply:
I had tried to separate these subscribe function into different component and the result is that only can subscribe successfully and others will be out of work. If you have other options to do so, please write the hint down. @frol